modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCachePartitionsRequest.java

```diff
@@ -77,97 +78,68 @@ public ClientCachePartitionsRequest(BinaryRawReader reader) {
             if (cacheDesc == null)
                 continue;
 
-            ClientCachePartitionAwarenessGroup grp = processCache(ctx, groups, cacheGroupIds, affinityVer, cacheDesc);
+            ClientCachePartitionAwarenessGroup grp = processCache(ctx, affinityVer, cacheDesc);
 
-            // Cache already processed.
             if (grp == null)
                 continue;
 
-            groups.add(grp);
-            cacheGroupIds.put(cacheDesc.groupId(), grp);
```

Comment:
Perhaps it worth to keep cacheGroupIds and check cacheGroupId for current cache before calculating it's mapping, to avoid double processing of mappings for cache groups from cacheIds.

Reply:
Discussed privately. Fixed some crutial points.